packages/api/src/modules/data-viz/paymentFlat/paymentFlat.service.ts

```diff
@@ -0,0 +1,46 @@
+import { ObjectId } from "mongodb"; // Import the ObjectId type
+import paymentFlatPort from "../../../dataProviders/db/paymentFlat/paymentFlat.port";
+import dataBretagneService from "../../providers/dataBretagne/dataBretagne.service";
+import chorusService from "../../providers/chorus/chorus.service";
+import PaymentFlatAdapter from "./paymentFlatAdapter";
+
+export class PaymentFlatService {
+    private async getAllDataBretagneData() {
+        const ministries = await dataBretagneService.getMinistriesRecord();
+        const programs = await dataBretagneService.findProgramsRecord();
+        const domainesFonct = await dataBretagneService.getDomaineFonctRecord();
+        const refsProgrammation = await dataBretagneService.getRefProgrammationRecord();
+        return { programs, ministries, domainesFonct, refsProgrammation };
+    }
+
+    public async upDatePaymentsFlatCollection(lastChorusObjectId: ObjectId) {
+        const { programs, ministries, domainesFonct, refsProgrammation } = await this.getAllDataBretagneData();
+
+        const chorusCursor = await chorusService.chorusCursorFind({ _id: { $gt: lastChorusObjectId } });
```

Comment:
On devrait avoir une méthode spécifique dans le repo pour le gt car la on fait du mongo dans un service

Reply:
car le service ne doit pas savoir que on utilise mongo c'est ça ? 
Du coup ça veut dire que la methode cursorFind dane le repository devrait être ou privé ou ne devrait pas pouvoir prendre query et projection comme paramètre ? car le format de ces paramètres depends de la base utilisé. Je me trompe ?

~~~typescript
public cursorFind(query: DefaultObject<unknown> = {}, projection: DefaultObject<unknown> = {}) {
        return this.collection.find(query, projection);
    }
~~~

Reply:
Hum en théorie oui, et même, on devrait avoir un cursor home made qui s'occupe de faire une couche entre le cursor mongo et le code metier

Reply:
done!